typing_extensions/src_py2/typing_extensions.py

```diff
@@ -89,3 +95,259 @@ def __subclasshook__(cls, C):
                     return True
             return NotImplemented
 
+
+def _collection_protocol(cls):
+    # Selected set of collections ABCs that are considered protocols.
+    name = cls.__name__
+    return (name in ('ABC', 'Callable', 'Awaitable',
+                     'Iterable', 'Iterator', 'AsyncIterable', 'AsyncIterator',
+                     'Hashable', 'Sized', 'Container', 'Collection', 'Reversible',
+                     'Sequence', 'MutableSequence', 'Mapping', 'MutableMapping',
+                     'AbstractContextManager', 'ContextManager',
+                     'AbstractAsyncContextManager', 'AsyncContextManager',) and
+            cls.__module__ in ('collections.abc', 'typing', 'contextlib',
+                               '_abcoll', 'abc'))
+
+
+class ProtocolMeta(GenericMeta):
+    """Internal metaclass for Protocol.
+
+    This exists so Protocol classes can be generic without deriving
+    from Generic.
+    """
+
+    def __new__(cls, name, bases, namespace,
+                tvars=None, args=None, origin=None, extra=None, orig_bases=None):
+        # This is just a version copied from GenericMeta.__new__ that
+        # includes "Protocol" special treatment. (Comments removed for brevity.)
+        if tvars is not None:
+            assert origin is not None
+            assert all(isinstance(t, TypeVar) for t in tvars), tvars
+        else:
+            tvars = _type_vars(bases)
+            gvars = None
+            for base in bases:
+                if base is Generic:
+                    raise TypeError("Cannot inherit from plain Generic")
+                if (isinstance(base, GenericMeta) and
+                        base.__origin__ in (Generic, Protocol)):
+                    if gvars is not None:
+                        raise TypeError(
+                            "Cannot inherit from Generic[...] or"
+                            " Protocol[...] multiple types.")
```

Comment:
I just use the same wording as in ``typing`` for consistency, but I actually agree that "multiple times" is more straightforward.